app/controllers/periods_controller.rb

```diff
@@ -66,4 +67,14 @@ def set_period
     def period_params
       params.require(:period).permit(:start, :end, :reports_count)
     end
+
+    # Create reports for all bench users for a given period
+    def create_reports_for_bench_users(period)
+      User.bench_users.each do |user|
+        Report.create!(
+          user: user,
+          period: period
+        )
+      end
```

Comment:
This feels something that needs to happen anytime a Period is created, does that sound right?

If so, then I think this fits better as [an active record callback](https://guides.rubyonrails.org/active_record_callbacks.html) instead of a method in a controller. Controllers are usually only responsible for methods that handle requests directly (using `render` or `redirect_to`) or methods that assist with that process by defining instance variables that are necessary for View templates.

But let's say someone created a `Period` in the `rails console`— `create_reports_for_bench_users` wouldn't run because `PeriodsController#create` never ran.

Using callbacks, you could configure Periods such that after any Period is created ActiveRecord will run a method for you. In this case you can make the method `create_reports_for_bench_users` and not remove this logic from the controller

Reply:
Hey Jelani, that is great feedback. Yes, this is supposed to happen any time a period is created. Active record callbacks are incredible - so helpful. I am kicking myself that I did not understand it until now. Makes sense to me to move it to the models file instead of the controller.

I have made an update on the same branch and pushed it. I then tried to and make a new pull request, but I didn't have that option. I am still trying to get used to pull requests and code review. But I think the changes updated in this pull request without having to make a new one. Let me know if how I did this works and is the best way to do it moving forward.

Reply:
@WilsonReese You can think of the pull request flow as something like:
1. You push a branch you plan to merge to GitHub and open a PR
2. You ask for review on said PR
3. Someone reviews the PR
4. You make changes based off of feedback and push them to the open PR
5. You ask for review again (steps 3-5 can sometimes repeat)
6. The reviewer approves all the changes you've made and you merge into `main` branch.

So in short, you did the right thing! Does that help?

Reply:
Yes that helps. It is good to know that pushing to an open PR works!